app/models/agents/user_location_agent.rb

```diff
@@ -68,7 +73,7 @@ def receive_web_request(params, method, format)
     def handle_payload(payload)
       location = Location.new(payload)
 
-      if location.present?
+      if (location.present? && (payload["accuracy"] < interpolated[:max_accuracy]))
```

Comment:
How about:

```ruby
if location.present? && (!interpolated[:max_accuracy].present? || payload["accuracy"] < interpolated[:max_accuracy])
```

It'd be great if you could also add a spec for this new behavior in user_location_agent_spec.rb.  If you have questions, I'm happy to help.